### ISSUE TITLE
AWI-CM3 install on KMA machine

### ISSUE THREAD

**Issue by sunseonlee** [esm-tools/esm_tools#966]
This is for AWI-CM3 installation on KMA machine (ICCP-AWI collaboration).

**Comment by pgierz** [comment]
Hi @sunseonlee, could you please give us some more information about what you are trying to do and how it is breaking so that we can help you fix it?

**Comment by sunseonlee** [comment]
We are trying to install awi-cm3 model on KMA machine (South Korea) to conduct global warming simulations for ICCP-AWI collaboration.

We are following the steps in https://esm-tools.readthedocs.io/en/v6.21.12/ten_steps.html.
When we ran ./install.sh on KMA machine, we got this error message.
![KakaoTalk_Photo_2023-05-24-15-10-49](https://github.com/esm-tools/esm_tools/assets/82024169/8988136d-b6e7-4035-893d-ce2977dd9215)

**Comment by JanStreffing** [comment]
So we have a general issue here in that this machine has no internet access. Therefore the requirements for esm_tools can not be downloaded for installation. We brainstormed two options for you:

1. Download each package required one by one from https://pypi.org and install manually via pip.
2. Use containers

We are currently trying to assess how much work each would be. Is the machine off the internet for the KMA admins too? It might be easier to have them install esm_tools if they have internet access.

**Comment by sunseonlee** [comment]
Internet is not available for KMA admins and support teams.

**Comment by pgierz** [comment]
For option 2 (which I prefer), it would be useful to know if your machine is able to use [singularity](https://docs.sylabs.io/guides/main/user-guide/). I can test on our own systems to ensure everything works, and the container file could just be copied across once it is ready.

**Comment by sunseonlee** [comment]
We will check if it is able to use singularity and let you know.

**Comment by mandresm** [comment]
Some additional and important information: **currently**, we did not support machines with no internet access. The reason for that is that our workflows were not developed on this type of machines, and giving support to ESM-Tools in these machines can be really challenging when things get tricky, as we cannot access them to see what's going on. Of course this policy can change but we need to be very careful with how we manage out human resources, anmd we need to think about how we are going to do that.

In the next days, we will be talking internally at AWI, to decide if we change this policy and we start to support off-line machines, how we would support that and which developments will be necessary for that. We will come back to you in this issue as soon as possible.

Please, keep us updated with the news about Singularity in KMA, and your progresses if you decide to give it a try to option 1 by yourself.

**Comment by sunseonlee** [comment]
Unfortunately, singularity is not available on KMA machine.

So, we will try to do option 1. 
Could you please tell me the list of package that should be downloaded and installed?

**Comment by pgierz** [comment]
Have a look at the setup.py. Like in any Python project, the requirements are listed there :-)

**Comment by sunseonlee** [comment]
Job scheduler of KMA machine is LSF (IBM), but we found that there are the files for pbs and slurm only in esm_tools/src/esm_runscrpts/. Should we make new files for LSF by ourselves? Could you please give us any comments on this?

**Comment by sunseonlee** [comment]
Here is error message that we have.

<img width="1098" alt="Screen Shot 2023-07-03 at 2 32 19 PM" src="https://github.com/esm-tools/esm_tools/assets/82024169/063ef92b-9328-4e8d-a963-f5a03cd3b5e6">

**Comment by pgierz** [comment]
Hi @sunseonlee,

Since we don't yet support that scheduler, you'd need to implement it yourselves. One of our team will be happy to help you with that once questions arise. Please make a separate issue and label it with enhancement so we have a separate space for discussion. We can save this issue for general problems and make a new one once more specific needs arise (like here)

@mandresm, @nwieters: I can provide first level assistance here.

**Comment by sunseonlee** [comment]
Hi @pgierz, 
Thank you very much for your comments.
I will open new issue for LSF system. 
One question: Is there any way to conduct awi-cm3 simulations without esm_tools?

**Comment by mandresm** [comment]
Hi @sunseonlee,

I see that both implementing lfs in ESM-Tools or running AWICM3 without ESM-Tools will be quite time consuming. Thomas Jung sent an email some time ago about organizing have a short telecon about it, so that we can agree on a strategy and understand what would be needed, and how we allocate the resources for this.

**Comment by sunseonlee** [comment]
Thanks @mandresm, and we will arrange the zoom call.

**Comment by nwieters** [comment]
Hi all,
are there any updates related to this issue?
Thank you

**Comment by JanStreffing** [comment]
The model is running on both KMA maschines DURU and NARU. However we can not use OMP threads. Jayeon Moon and myself are still working on this. I also would like to see a merge back into release onces everything is cleaned up, so that in case we ever have another LSF maschine we have something to build upon. 

As such I would suggest to leave the issue open, even though I must say it's been emails rather than communication here for the past 3 months...

**Comment by mandresm** [comment]
I would love to see as well a merge back from the LSF work into `release` :)
